runners/spark/src/main/java/org/apache/beam/runners/spark/util/SideInputReaderUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.spark.util;
+
+import java.util.Map;
+import org.apache.beam.runners.core.SideInputReader;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.WindowingStrategy;
+
+/**
+ * Utility class for creating and managing side input readers in the Spark runner.
+ *
+ * <p>This class provides factory methods to create appropriate {@link SideInputReader}
+ * implementations based on the execution mode (streaming or batch) to optimize side input access
+ * patterns.
+ */
+public class SideInputReaderUtils {
```

Comment:
From Beam naming convention it sounds this is a "SideInputReaderFactory". And the static method can be named after "create"

Reply:
@Abacn 
Thanks for pointing out the naming convention. I'll rename it to SideInputReaderFactory with a static create method to better align with Beam's conventions.